### PR TITLE
Add endpoint manager tests & small clean-ups

### PR DESCRIPTION
**Pull request by aanm**
Read per commit basis

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6170)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

File: pkg/endpoint/id/id.go [hunk 1]
```diff
@@ -67,20 +67,13 @@ func ParseCiliumID(id string) (int64, error) {
...
-// ParseID parses specified id and returns normalized id as string.
-func ParseID(id string) (PrefixType, string, error) {
+// ParsePrefix parses the PrefixType from the given id that contains the prefix
+// and an ID. Returns a PrefixType and the ID. An error is returned if the prefix
+// is not known by the constants of valid prefixes.
+func ParsePrefix(id string) (PrefixType, string, error) {
 	prefix, eid := SplitID(id)
 	switch prefix {
-	case CiliumLocalIdPrefix:
-		if _, err := ParseCiliumID(id); err != nil {
```
Comment (tgraf) [inline pkg/endpoint/id/id.go:78]: Is it safe to remove this validation?

**Review by tgraf — CHANGES_REQUESTED** [review]

File: pkg/endpoint/id/id.go (hunk 1 above)
Comment (aanm) [reply pkg/endpoint/id/id.go:78]: Yes, there are only 2 callers of that function, of them doesn't even use the returned `eid`, the other executes ParseCiliumID afterwards

**Review by tgraf — APPROVED** [review]

**Pull request merged**